src/OpenApi/src/Comparers/OpenApiAnyComparer.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.OpenApi.Any;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal sealed class OpenApiAnyComparer : IEqualityComparer<IOpenApiAny>
+{
+    public static OpenApiAnyComparer Instance { get; } = new OpenApiAnyComparer();
+
+    public bool Equals(IOpenApiAny? x, IOpenApiAny? y)
+    {
+        if (x is null && y is null)
+        {
+            return true;
+        }
+        if (x is null || y is null)
+        {
+            return false;
+        }
+
+        return GetHashCode(x) == GetHashCode(y);
```

Comment:
The two objects might not be the same if equals is implemented like this. Won't that be a problem?

Reply:
I think the typical expectation is that types with the same hash codes _might_ be equal whereas types with different hash codes are definitely _not_ equal.

Assuming I've done my implementation correctly, `GetHashCode` for `OpenApiSchema` and should return a unique value for each equivalent schema object (e.g. the bucket size is always 1).

Reply:
When working with hashcodes I don't think you can guarantee the bucket size is 1 given the nature of the objects. The data in the object is being reduced down to a 2^32 number. There are infinite possible ways of defining schemas that are reduced to a hashcode so it seems like at some point there must be hash collisions.

I question what is happening here, but I also don't feel certain enough in my understanding of hash tables to be sure. Maybe ping an expert in this area?

Reply:
Your questions here are legit. After some rubber ducking, it appears there is a gap here when evaluating `OpenApiByte` and `OpenApiBinary` types since they encapsulate a byte area that we were evaluating correctly.  I did some hardening on the comparer implementations in https://github.com/dotnet/aspnetcore/pull/56175/commits/623fa19e5cf87be6e2991fcd9710b954df84bf07